src/walletd/runtime.rs

```diff
@@ -360,8 +360,18 @@ impl Runtime {
                 error!("Msg and source don't have same swap_id, ignoring...");
                 return Ok(());
             }
+
+            Request::Protocol(msg)
+                if req_swap_id.is_some() && Some(msg.swap_id()) == req_swap_id => {}
             // TODO enter farcasterd messages allowed
-            _ => {}
+            _ if source == ServiceId::Farcasterd => {}
```

Comment:
This overrides the first match arm. Either remove the specific selection of the TakerCommit protocol message or this arm here.